### ISSUE TITLE
Confirmation before resetting editor

### ISSUE THREAD

**Issue by adhameer** [ocaml-sf/learn-ocaml#187]
The Reset button in the editor is very easy to hit accidentally on a touch screen and there is no way to undo it. A confirmation dialog would be very useful.

**Comment by erikmd** [comment]
@adhameer indeed, the current behavior of `Reset` is somewhat risky, I was about to open the very same feature request :)
Actually, if after clicking on `Reset` one immediately types `Ctrl+Z` in the buffer, one may undo the reset... but this seems not possible if any other buffer modification occurs after the `Reset` (including automatic re-indent).
